routing/router.go

```diff
@@ -1873,38 +1873,40 @@ func (r *ChannelRouter) tryApplyChannelUpdate(rt *route.Route,
 // error type, this error is either the final outcome of the payment or we need
 // to continue with an alternative route. This is indicated by the boolean
 // return value.
-func (r *ChannelRouter) processSendError(rt *route.Route, sendErr error) (
-	bool, channeldb.FailureReason) {
+func (r *ChannelRouter) processSendError(rt *route.Route, sendErr error) (bool,
+	channeldb.FailureReason) {
 
-	// If the failure message could not be decrypted, attribute the failure
-	// to our own outgoing channel.
-	//
-	// TODO(joostager): Penalize all channels in the route.
-	if sendErr == htlcswitch.ErrUnreadableFailureMessage {
-		sendErr = &htlcswitch.ForwardingError{
-			FailureSourceIdx: 0,
-			FailureMessage:   lnwire.NewTemporaryChannelFailure(nil),
-		}
-	}
+	var (
+		failureMessage   lnwire.FailureMessage
+		failureSourceIdx *int
+	)
 
-	// If an internal, non-forwarding error occurred, we can stop trying.
-	fErr, ok := sendErr.(*htlcswitch.ForwardingError)
-	if !ok {
-		return true, channeldb.FailureReasonError
-	}
+	if sendErr != htlcswitch.ErrUnreadableFailureMessage {
```

Comment:
would this be easier to follow using a switch statement?

Reply:
not sure exactly how to convert to a switch, but i took another look at this part and simplified it in another way.